mage_ai/data_preparation/models/pipeline.py

```diff
@@ -434,11 +434,17 @@ def delete_block(self, block, widget=False):
         if block.uuid not in mapping:
             raise Exception(f'Block {block.uuid} is not in pipeline {self.uuid}.')
         if len(block.downstream_blocks) > 0:
-            downstream_block_uuids = [b.uuid for b in block.downstream_blocks]
-            raise Exception(
-                f'Blocks {downstream_block_uuids} are depending on block {block.uuid}'
-                '. Please remove the downstream blocks first.'
-            )
+            downstream_block_uuids = [
+                b.uuid for b in block.downstream_blocks if b.type != BlockType.CHART
+            ]
+            if len(downstream_block_uuids) > 0:
+                raise Exception(
+                    f'Block(s) {downstream_block_uuids} are depending on block {block.uuid}'
+                    '. Please remove the downstream blocks first.'
+                )
+            for downstream_block in block.downstream_blocks:
+                downstream_block.delete()
```

Comment:
we update pipeline config file (self.save()) every time we delete a block. for this case, i think we can delay updating the file after deleting all downstream blocks. so maybe add a param called `commit` to the delete method, set it to False when deleting downstream blocks.

Reply:
Added commit parameter to `delete`  and `delete_block`. Of note, when checking if the block is used in other pipelines, I had to filter out the current pipeline manually since the pipeline config is not immediately updated.